src/main/java/org/mule/extension/db/internal/domain/connection/oracle/OracleDbConnection.java

```diff
@@ -161,27 +167,44 @@ protected void resolveLobs(String typeName, Object[] attributes, StructAndArrayT
 
   @Override
   protected Map<Integer, ResolvedDbType> getLobFieldsDataTypeInfo(String typeName) throws SQLException {
-    Map<Integer, ResolvedDbType> dataTypes = new HashMap<>();
 
-    Optional<String> owner = getOwnerFrom(typeName);
-    String type = getTypeSimpleName(typeName);
+    if (this.resolvedDbTypesCache.containsKey(typeName)) {
+      if (logger.isDebugEnabled()) {
+        logger.debug("Returning chached LobFieldsDataTypeInfo");
+      }
+      return resolvedDbTypesCache.get(typeName);
+    }
+    if (logger.isDebugEnabled()) {
+      logger.debug("Obtaining LobFieldsDataTypeInfo");
+    }
+    synchronized (resolvedDbTypesCache) {
+      if (this.resolvedDbTypesCache.containsKey(typeName)) {
+        return resolvedDbTypesCache.get(typeName);
+      }
 
-    String query = QUERY_TYPE_ATTRS + (owner.isPresent() ? QUERY_OWNER_CONDITION : "");
+      Map<Integer, ResolvedDbType> dataTypes = new HashMap<>();
```

Comment:
wasn't this a ConcurrentHashmap on the original PR?

Reply:
No, this was the original code to store the datatypes. It was filled at every call. The original PR had a ConcurrentHashMap in the OracleDbConnectionProvider.